plugins/hls-class-plugin/src/Ide/Plugin/Class/CodeAction.hs

```diff
@@ -207,6 +213,9 @@ isClassNodeIdentifier ident = (isNothing . identType) ident && Use `Set.member`
 isClassMethodWarning :: T.Text -> Bool
 isClassMethodWarning = T.isPrefixOf "• No explicit implementation for"
 
+isInstanceMissingError :: T.Text -> Bool
+isInstanceMissingError = T.isPrefixOf "• No instance for"
```

Comment:
What if this error comes from something not in an instance definition?
If I guess correctly, this will give error responses, which is not so desirable.
I think it's probably better to have a more specific way to detect this "invalid default impl" case.

Reply:
Hmm, yes this produces logs like `Warning | class: InternalError: No Identifier found`. I am not sure what is a better way to detect this error, here are two errors:

```
• No instance for (Test1 X)
    arising from a use of ‘DefaultImplementation.$dmf’
• In the expression: DefaultImplementation.$dmf @(X)
  In an equation for ‘f’: f = DefaultImplementation.$dmf @(X)
  In the instance declaration for ‘Test X’
```

```
• No instance for (Show a) arising from a use of ‘show’
  Possible fix:
    add (Show a) to the context of
      the type signature for:
        foo :: forall a. a -> String
• In the expression: show
  In an equation for ‘foo’: foo = show
```

generated from file:

```
{-# LANGUAGE DefaultSignatures #-}
module DefaultImplementation where

class Test1 a where

class Test a where
  f :: a -> a
  default f :: Test1 a => a -> a
  f = id

data X = X | Y


foo :: a -> String
foo = show

instance Test X where
```

Do you think trying to see if there is no "Possible fix" would be better? Or should the plugin just start expecting this other case and not throw errors when it doesn't find any class identifier?

Reply:
How about looking for the "In instance" part? That sounds more promising. And surely it'd be better if it doesn't report HLS errors for non-default instance resolution GHC errors.

Reply:
BTW, how does this work for something like
```hs
data X = X | Y

class Test a where
  f :: a -> String

instance Test X where
  f = show
```
?